src/identity/identity-wallet.ts

```diff
@@ -555,6 +612,106 @@ export class IdentityWallet implements IIdentityWallet {
     };
   }
 
+  /**
+   *
+   * creates Ethereum based identity
+   *
+   * @param {IdentityCreationOptions} opts - options for the creation of the identity
+   * @returns `{did,credential>}` - returns did and Auth BJJ credential
+   */
+  private async createEthereumIdentity(
+    opts: IdentityCreationOptions
+  ): Promise<{ did: DID; credential: W3CCredential }> {
+    opts.method = opts.method ?? DidMethod.Iden3;
+    opts.blockchain = opts.blockchain ?? Blockchain.Polygon;
+    opts.networkId = opts.networkId ?? NetworkId.Mumbai;
+    opts.seed = opts.seed ?? getRandomBytes(32);
+
+    const proofService = opts.proofService;
+    const ethSigner = opts.ethSigner;
+
+    const currentState = ZERO_HASH; // In Ethereum identities we don't have an initial state with the auth credential
+
+    const didType = buildDIDType(opts.method, opts.blockchain, opts.networkId);
+
+    const keyIdEth = await this._kms.createKeyFromSeed(KmsKeyType.Secp256k1, opts.seed);
+    const pubKeyHexEth = (await this._kms.publicKey(keyIdEth)).slice(2); // 04 + x + y (uncompressed key)
+    // Use Keccak-256 hash function to get public key hash
+    const hashOfPublicKey = keccak256(Buffer.from(pubKeyHexEth, 'hex'));
+    // Convert hash to buffer
+    const ethAddressBuffer = Buffer.from(hashOfPublicKey, 'hex');
+    // Ethereum Address is '0x' concatenated with last 20 bytes
+    // of the public key hash
+    const ethAddr = ethAddressBuffer.slice(-20);
+    const genesis = genesisFromEthAddress(ethAddr);
+    const identifier = new Id(didType, genesis);
+    const did = DID.parseFromId(identifier);
```

Comment:
let's move this to utils function and do a proper test for it

Reply:
Moved and test created